File: Database/py/mongo_connection.py
import mysql.connector
from pymongo import MongoClient
from sql_kliens import *
from mongo_kliens import *



# Adatbázis kulcs alapján történő választás
def choose_database(key,username,email,password):
    if key == "sql":
        sql_open_con()
        # POST
        # ÚJ KLIENS LÉTREHOZÁSA
        sql_post_new_client(username, email, password)

        # GET
        # KLIENS LEKÉRÉSE ID ALAPJÁN
        id = 1
        print(sql_get_client_by_id(id))

        # KLIENS LEKÉRÉSE EMAIL ALAPJÁN
        email = 'jancsi@ljb.vd'
        print(sql_get_client_by_email(email))

        # Szavazat rögzítése

        voter_name = "Lajos"
        vote = "Meg semmit."
        kerdes = "Mit ittal ma?"

        sql_record_vote(voter_name,vote,kerdes)

        result = sql_get_vote_by_voter("John Doe")
        print(result)

    elif key == "mongodb":
        # MongoDB adatbázis kapcsolat inicializálása
        mongo_open_con()
        # POST
        # ÚJ KLIENS LÉTREHOZÁSA
        mongo_post_new_client(username, email, password)

        # GET
        # KLIENS LEKÉRÉSE ID ALAPJÁN
        id = 1
        print(mongo_get_client_by_id(id))

        # KLIENS LEKÉRÉSE EMAIL ALAPJÁN
        email = 'jancsi@ljb.vd'
        print(mongo_get_client_by_email(email))
        # Szavazat rögzítése

        voter_name = "John Doe"
        vote = "A"

        mongo_record_vote(voter_name,vote)

        result = mongo_get_vote_by_voter("John Doe")
        print(result)
    else:
        print("Érvénytelen adatbázis kulcs.")
        return None

# Példa adatbázis használatra kulcs alapján
# database_key = "sql"

File: Database/mongo_connection.php
<?php
include 'db_connection.php';
// Adatbázis kulcs alapján történő választása
function chooseDatabase($key) {
    if ($key === "sql") {
        // SQL adatbázis kapcsolat inicializálása
        $dbhost = "localhost";
        $dbuser = "aporka";
        $dbpass = "nehezjelszo";
        $db = "ORProjekt_Szavazatok";
        $conn = new mysqli($dbhost, $dbuser, $dbpass,$db) or die("Connect failed: %s\n". $conn -> error);

        if(!($conn -> error)){
            echo "Succesfuly connected to the database!";
        }
        

        return $conn;
    } elseif ($key === "mongodb") {
        // MongoDB adatbázis kapcsolat beállítása
        $mongoClient = new MongoDB\Client("mongodb://localhost:27017");
        $mongoDB = $mongoClient->selectDatabase("szavazatok");

        // Szavazat rögzítése
        
        $voterName = "John Doe";
        $vote = "Option A";

        recordVote($voterName, $vote);

        return $mongoDB; 
    } else {
        echo "Érvénytelen adatbázis kulcs.";
        return null;
    }
}

// Példa adatbázis használatra kulcs alapján
$databaseKey = "sql"; 

$database = chooseDatabase($databaseKey);

function recordVote($voterName, $vote) {
    global $mongoDB;
    
    $collection = $mongoDB->votes;
    
    $document = array(
        "voterName" => $voterName,
        "vote" => $vote
    );
    
    $collection->insertOne($document);
    
    echo "Szavazat sikeresen rögzítve.";
}

function CloseCon($conn){
    $conn -> close();
}
?>
